rtest.py

```diff
@@ -65,6 +65,7 @@
 PASS = 'PASS'
 FAIL = 'FAIL'
 ERROR = 'ERROR'
+SKIP = 'SKIP'
 
```

Comment:
Hmm, how about fail is red, and error is bold red?

Reply:
Btw, there should **never** be a new SKIP, right?.. Unless one puts it in by hand?

Reply:
Or if one creates a broken test by forgetting to commit something (choices, gold profile, etc.) or doesn't add it to the index. Recall there are "disabled" skips and "incomplete" skips.

Reply:
I can’t find a reference now, but I believe there is a precedent for using blue to indicate skipped tests.

Reply:
> Or if one creates a broken test by forgetting to commit something (choices, gold profile, etc.) or doesn't add it to the index. Recall there are "disabled" skips and "incomplete" skips.

Hmm OK, so, they are potentially distracting, then. I am thinking from the point of view, you are running tests, and it is easy to notice something's off if you used to have all green, zero failures of any kind, and now suddenly you've got something. Having 23 skipped tests (what we have now) is then not ideal because now you have those 23 items to think about, too. Is it still 23, or did it become 24? Has it always been 24? etc. Not a clean situation when you know for sure things are fine and nothing changed...

Reply:
I guess at least SKIP gives you confidence that it is not some  difference in the MRS.

Actually, wouldn't it be better to treat disable and incomplete as separate? The incomplete ones I think need to be either deleted or completed... They are broken.

Reply:
To clarify: I think getting used to some 23 skipped tests (some of which may be broken) flashing in front of you in the output leads to just becoming insensitive to that information. So, the situation becomes similar to keeping some failed tests around which we did for a while with information structure, and it is not ideal because you sort of become accustomed to it and start ignoring this kind of information altogether (at least to a degree). So I want to have a situation where there is confidence that nothing changed between two runs, and that everything that is skipped is skipped because it was disabled by hand. The rest of them I think should be either removed or indexed and disabled, and the new ones which were added improperly should be ERRORING?

Reply:
> I believe there is a precedent for using blue to indicate skipped tests in Python.

I was following `pytest`'s convention, which uses yellow (and red for fail/error). If blue makes more sense for people, that's fine.

> Hmm OK, so, they are potentially distracting, then. [...]

I understand it may make you feel uneasy to see something not green, but these have been here for years. The change is just making them visible so we can act on them. These have been here for some time (see my thread on the matrix-dev list from about a year ago titled "Auditing the Matrix regression test system").

> I guess at least SKIP gives you confidence that it is not some difference in the MRS.

A skipped test is not a failed test, nor is it a passed test. I don't think it should grant one any confidence nor particular unease.

> Actually, wouldn't it be better to treat disabled and incomplete as separate?

I don't think so. If we make them errors, then the tests fail, and also they'd be lumped imperfectly into another category: things that raised actual errors. From the point of view of running the regression tests, the result of the test is that it was skipped, not that it passed, failed, or raised an error. If you want to see *why* something was skipped, `rtest.py --list --skipped --verbose` (or `rtest.py -lv --skipped` for short) gives you that information.

> I think getting used to some 23 skipped tests (some of which may be broken) flashing in front of you in the output leads to just becoming insensitive to that information [...]

I sympathize with that, but the point of skipped tests (instead of ignoring known failures) is that we've decided that they are not counted as failures. They're more like reminders. Ideally they would be fixed and re-enabled, but we can't make everything perfect at once.

Reply:
> If we make them errors, then the tests fail, and also they'd be lumped imperfectly into another category: things that raised actual errors.

I think I don't see how an incompletely added (or incompletely removed) test is not an actual error?

Reply:
It's a different category of error, then? The "error" tests are something like an exception raised by Python, or ACE crashing. These are just not run because some of the necessary files are missing. I guess we could force it to an actual error by trying to run the tests with missing files anyway, but I don't like that option.

Reply:
Yes, I think it is a kind of error, different internally from ACE crashing (but ultimately not that different from Python raising an exception, which is exactly what would happen if the test were run).

> I guess we could force it to an actual error by trying to run the tests with missing files anyway, but I don't like that option.

Why not? What would be a scenario in which it makes sense to just keep a test like this in its incomplete form? I don't like that they will be lumped together with the ones which we explicitly disabled by hand. If I by accident delete some part of the test, it will become a SKIP and I won't even notice because I will be assuming that it had been disabled. 

Remembering exactly which tests were in which state in the previous run is a great skill to have and I think most people do not have it. It is easy to not notice a change in a situation like this. So, it is not that I simply want to always see all green :). I just want to avoid a pitfall here (I've been working with hundreds of tests for a long time now).

Reply:
> Why not? What would be a scenario in which it makes sense to just keep a test like this in its incomplete form? I don't like that they will be lumped together with the ones which we explicitly disabled by hand. If I by accident delete some part of the test, it will become a SKIP and I won't even notice because I will be assuming that it had been disabled.

Ok, thank you for persisting on this topic. I'm coming around to your point of view. Ultimately, yes, these are broken tests and we probably should flag them as errors if new broken tests are created somehow. One reason they are skipped now is because they had been ignored (but not explicitly skipped) for so long, and when I first created the rtest.py replacement for the old regression test system, I wanted it to replicate the previous behavior while making it easy to discover the now-visible existing problems. If we're ready to deal with all the broken tests (#499) then I'm prepared to make these proper errors.

Reply:
Yes, I will deal with the broken tests (as part of https://github.com/delph-in/matrix/pull/499).